xclim/indices/fwi.py

```diff
@@ -108,79 +138,103 @@
 )
 
 
-def day_length(lat):
-    """Return the average day length by month within latitudinal bounds."""
-    lat_bnds = (-90, -30, -15, 15, 33, 90)
-    i = np.digitize(lat, lat_bnds) - 1
-    return DAY_LENGTHS[i]
-
+@jit
+def day_length(lat, mth):  # pragma: no cover
+    """Return the average day length for a month within latitudinal bounds."""
+    if -30 > lat >= -90:
+        dl = DAY_LENGTHS[0, :]
+    elif -15 > lat >= -30:
+        dl = DAY_LENGTHS[1, :]
+    elif 15 > lat >= -15:
+        return 9
+    elif 30 > lat >= 15:
+        dl = DAY_LENGTHS[3, :]
+    elif 90 >= lat >= 30:
+        dl = DAY_LENGTHS[4, :]
+    return dl[mth - 1]
+
+
+@jit
+def day_length_factor(lat, mth):  # pragma: no cover
+    """Return the day length factor"""
+    if -15 > lat >= -90:
+        dlf = DAY_LENGTH_FACTORS[0, :]
+    elif 15 > lat >= -15:
+        return 1.39
+    elif 90 >= lat >= 15:
+        dlf = DAY_LENGTH_FACTORS[2, :]
+    return dlf[mth - 1]
+
+
+@vectorize
+def fine_fuel_moisture_code(t, p, w, h, ffmc0):  # pragma: no cover
```

Comment:
Would it break functionality if these were kwargs?

Reply:
No, but I believe the compiler wants positional arguments, so we wouldn't gain much.

Reply:
Got it.